### PR TITLE
docs: add wise-ft to clip finetuning

### PR DESCRIPTION
**Pull request by bwanglzu**
In this PR, we start to support WiSE-FT fine-tuning for models with a strong zero-shot capability, to merge weights between pre-trained model and zero-shot model. Currently all CLIP-family models are supported. Functionality has been implemented inside core. To make use of it:

```diff
import finetuner

run = finetuner.fit(
     model='ViT-B-32#openai',
     ...,
     loss = 'CLIPLoss',
-    callbacks=[],
+    callbacks=[WiSEFTCallback(alpha=0.4)],
)
```

Review documentation change [here](https://ft-docs-wise-ft--jina-docs.netlify.app/docs-wise-ft/tasks/text-to-image/#advanced-wise-ft)

---

- [x] This PR references an open issue
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

**Review by LMMilliken — APPROVED** [review]

File: docs/tasks/text-to-image.md [hunk 1]
```diff
@@ -163,6 +163,38 @@ In case you set `to_onnx=True` when calling `finetuner.fit` function,
...
\ No newline at end of file
+WiSE-FT, proposed by Mitchell et al. in [Robust fine-tuning of zero-shot models](https://arxiv.org/abs/2109.01903),
+has been proven to be an effective way for fine-tuning models with a strong zero-shot capability,
+such as CLIP.
+As was introduced in the paper:
+
+> Large pre-trained models such as CLIP or ALIGN offer consistent accuracy across a range of data distributions when performing zero-shot inference (i.e., without fine-tuning on a specific dataset). Although existing fine-tuning methods substantially improve accuracy on a given target distribution, they often reduce robustness to distribution shifts. We address this tension by introducing a simple and effective method for improving robustness while fine-tuning: ensembling the weights of the zero-shot and fine-tuned models (WiSE-FT).
+
+Finetuner allows you to apply WiSE-FT easily,
+all you need to do is use `WiSEFTCallback`.
```
Comment (gmastrapas) [inline docs/tasks/text-to-image.md:177]: use the

File: docs/tasks/text-to-image.md [hunk 2]
```diff
@@ -163,6 +163,38 @@ In case you set `to_onnx=True` when calling `finetuner.fit` function,
...
+Finetuner will trigger the callback when fine-tuning job finished and merge the weights between the pre-trained model and the fine-tuned model:
+
+```diff
+import finetuner.callbakcs import WiSEFTCallback
```
Comment (gmastrapas) [inline docs/tasks/text-to-image.md:181]: from

File: docs/tasks/text-to-image.md [hunk 3]
```diff
@@ -163,6 +163,38 @@ In case you set `to_onnx=True` when calling `finetuner.fit` function,
...
+run = finetuner.fit(
+    model='ViT-B-32#openai',
+    ...,
+    loss='CLIPLoss',
+-   callbacks=[],
++   callbacks=[WiSEFTCallback(alpha=0.5)],
+)
+```
+
+The value you set to `wise_ft` should be greater equal than 0 and less equal than 1:
```
Comment (gmastrapas) [inline docs/tasks/text-to-image.md:192]: alpha not wise_ft

File: docs/tasks/text-to-image.md [hunk 4]
```diff
@@ -163,6 +163,38 @@ In case you set `to_onnx=True` when calling `finetuner.fit` function,
...
+
++ if `wise_ft` is a float between 0 and 1, we merge the weights between the pre-trained model and the fine-tuned model.
```
Comment (gmastrapas) [inline docs/tasks/text-to-image.md:194]: same

File: docs/tasks/text-to-image.md [hunk 5]
```diff
@@ -163,6 +163,38 @@ In case you set `to_onnx=True` when calling `finetuner.fit` function,
...
++ if `wise_ft` is 0, the fine-tuned model is identical to the pre-trained model.
```
Comment (gmastrapas) [inline docs/tasks/text-to-image.md:195]: same

File: docs/tasks/text-to-image.md [hunk 6]
```diff
@@ -163,6 +163,38 @@ In case you set `to_onnx=True` when calling `finetuner.fit` function,
...
++ if `wise_ft` is 1, the pre-trained weights will not be utilized.
```
Comment (gmastrapas) [inline docs/tasks/text-to-image.md:196]: same

File: docs/walkthrough/using-callbacks.md [hunk 7]
```diff
@@ -126,3 +126,17 @@ The early stopping callback triggers at the end of training and evaluation batch
 - `minimum_delta`: The minimum amount of improvement that a model can have over the previous best model to be considered worthwhile, zero by default, meaning that the training will not stop early unless the performance starts to decrease
 - `patience`: The number of consecutive rounds without improvement before the training is stopped, two by default.
 - `baseline`: an optional parameter that is used to compare the model's score against instead of the best previous model when checking for improvement. This baseline does not get changed over the course of a run.
+
+
+## WiSTFTCallback
```
Comment (gmastrapas) [inline docs/walkthrough/using-callbacks.md:131]: WiSEFTCallback

**Review by gmastrapas — APPROVED** [review]

File: docs/walkthrough/using-callbacks.md (hunk 7 above)
Comment (bwanglzu) [reply docs/walkthrough/using-callbacks.md:131]: ah good catch, i hope i didn't make it wrong in core
